contracts/modules/LidoWrapHandler.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import '@yield-protocol/utils-v2/contracts/token/IERC20.sol';
+import '@yield-protocol/utils-v2/contracts/token/TransferHelper.sol';
+import '../LadleStorage.sol';
+import '../oracles/lido/IWstETH.sol';
+
+/// @dev A contract to handle wrapping & unwrapping of stETH
+contract LidoWrapHandler {
+    using TransferHelper for IERC20;
+    using TransferHelper for IWstETH;
+
+    IWstETH public immutable wstETH;
+    IERC20 public immutable stETH;
+
+    /// @dev The amount of stETH wrapped
+    event Wrapped(address to, uint256 amount);
+
+    /// @dev The amount of wstETH unwrapped
+    event Unwrapped(address to, uint256 amount);
+
+    constructor(IWstETH wstETH_, IERC20 stETH_) {
+        wstETH = wstETH_;
+        stETH = stETH_;
+    }
+
+    /// @dev Wrap stEth held by this contract and forward it to the 'to' address
+    function wrap(address to) external returns (uint256 wrappedAmount) {
+        uint256 stEthTransferred = stETH.balanceOf(address(this));
+        require(stEthTransferred > 0, 'No stETH to wrap');
+        stETH.approve(address(wstETH), stEthTransferred);
```

Comment:
You can do this in the constructor and save some gas for each user.

`stEth_.approve(address(wstEth_), type(uint256).max)`

It is ok because this contract is not expected to hold any tokens between transactions, to that max approve can't be used to steal anything.